src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+
+  function onKeyDown(event) {
+    switch (event.key) {
+      case 'Escape':
+        setExpanded(false);
+        break;
+      default:
+    }
+  }
+
+  useDropdownClose(wrapperRef, expanded, onDropdownClose);
+
+  useEffect(() => {
+    setValue(props.value);
+  }, [props.value.map(val => JSON.stringify(val)).join(',')]);
+
+  useEffect(() => {
+    props.onChange({ target: selfRef.current });
+  }, [value.map(val => JSON.stringify(val)).join(',')]);
+
+  useImperativeHandle(selfRef, () => ({
+    get dirty() {
+      return props.value.map(val => JSON.stringify(val)).join(',') !== this.value.map(val => JSON.stringify(val)).join(',');
+    },
+    id: props.id,
+    name: props.name,
+    get value() {
+      return value;
+    },
+  }));
+
+  return (
+    <div className={classNames.container} id={props.id} ref={wrapperRef}>
+      <FormControl
+        label={props.label}
+        labelId={ids.label}
+        id={ids.textbox}
+        error={validationMessage}
+        onKeyDown={onKeyDown}
+      >
+        <div role="presentation" className={classNames.formControlChildrenContainer} onClick={onClick}>
+          <TagList
+            className={classNames.tagList}
+            labelledBy={ids.label}
+            refs={valueRefs}
+          >
+            {value.length !== 0 &&
+              props.tagChildren ?
+              props.tagChildren(value, valueRefs, onOptionRemove) :
+              value.map((val, index) => (
+                <Tag
+                  defaultActive={index === 0}
+                  id={`${props.id}-option-${props.optionIDGetter(getOption(val))}`}
+                  key={props.optionIDGetter(getOption(val))}
+                  onRemove={onOptionRemove}
+                  readOnly={props.readOnly}
+                  ref={valueRefs[index]}
+                >
+                  {props.optionLabelGetter(
+                    props.options.find(option => props.optionIDGetter(option) === props.optionIDGetter(getOption(val))),
+                  )}
+                </Tag>
+              ))
+            }
+            <input
+              aria-autocomplete="list"
+              aria-controls={ids.listbox}
+              aria-describedby={`${ids.emptyMessage}`}
+              aria-expanded={expanded}
+              aria-haspopup="listbox"
+              aria-labelledby={ids.label}
+              autoComplete="off"
+              role="combobox"
+              className={classNames.input}
+              id={ids.textbox}
+              onBlur={onAutocompleteBlur}
+              onChange={onAutocompleteChange}
+              placeholder={value.length === 0 ? props.placeholder : undefined}
+              readOnly={props.readOnly}
+              required={props.required ? value.length === 0 : false}
+              ref={autocompleteRef}
+              value={autocompleteValue}
+            />
+          </TagList>
+          <FormControlIcons validationMessage={props.validationMessage} className={classNames.iconsContainer}>
```

Comment:
For some odd reason, we're missing the invalid icon. 
![image](https://user-images.githubusercontent.com/841084/118866047-7572e000-b896-11eb-93af-9902453c10d4.png)

Reply:
While we're talking about an the invalid state, we need to also make the border red. See previous implementation:

![image](https://user-images.githubusercontent.com/841084/118866181-963b3580-b896-11eb-8326-87c40c0a746b.png)

Reply:
Interesting, this may be an unintended regression. I'll check it out and get it working properly.